WordPress/gutenberg
# FontSizePicker: Rewrite unit tests to use userEvent and be more comprehensive

## What?
Rewrites the `FontSizePicker` unit tests to use `userEvent` where possible and be more comprehensive.

## Why?
I want to make @ciampo happy 🙂 see https://github.com/WordPress/gutenberg/pull/44891#pullrequestreview-1142800395.

## How?
I looked at the `FontSizePicker` component storybook, listed all of the functionality that I could think of, and then wrote new tests from scratch that covered everything that I listed.

## Testing Instructions
`npm run test`, obviously, but the more important thing is to review the code and check that I haven't missed anything.

## Screenshots or screencast
